### ISSUE TITLE
Feature request: Shoryuken integration

### ISSUE THREAD

**Issue by steveh** [DataDog/dd-trace-rb#538]
Shoryuken is a Ruby worker for Amazon SQS.

It has [middleware](https://github.com/phstc/shoryuken/wiki/Middleware) based on Sidekiq, with a slightly different signature:

```ruby
class MyMiddleware
  def call(worker_instance, queue, sqs_msg, body)
    puts 'Before work'
    yield
    puts 'After work'
  end
end
```

An important thing to note is that Shoryuken can be run in a batch mode, in which case `sqs_msg` and `body` will be arrays, see [this test](https://github.com/phstc/shoryuken/blob/master/spec/shoryuken/middleware/server/auto_delete_spec.rb#L30) for an example.

It uses the AWS SQS API underneath, which is already partially tracked by ddtrace.

**Comment by delner** [comment]
Cool, this is really helpful detail @steveh ! I'm going to add this to our backlog now, so we can get this scheduled. Thanks for bringing this one to our attention!

**Comment by kawamanza** [comment]
Hello @delner, I had locally implemented this middleware to allow monitoring on Shoryuken workers recently. It's working fine and I thought about sharing this middleware with the community. Did you start implementing this middleware?

**Comment by kawamanza** [comment]
Done :point_up:

**Comment by delner** [comment]
@steveh @kawamanza we merged #626 which adds support for this. It should roll with the 0.18.0 release. Closing this issue; thank you!
